graph-builder/src/graph.rs

```diff
@@ -142,5 +150,101 @@ fn create_graph(
             })
         })?;
 
+    graph.prune_abstract();
+
     Ok(graph)
 }
+
+/// Removes and retrieves the metadata value of the given `key` which is prefixed
+/// by `filter`
+fn remove_prefixed_label(
+    filter: &str,
```

Comment:
Nit: I would personally re-order arguments so that the `&mut` comes first and filter+key are next to each other.

Reply:
Just for future reference: we can do that ordering for new methods but not when changing mutability